### PR TITLE
change visualize json to get json from rosparam

### PR DESCRIPTION
**Pull request by yuyu2172**

### CODE REVIEW

**Comment by yuyu2172** [comment]
You can change rosparam after starting the node.

**Comment by knorth55** [comment]
👍

**Comment by wkentaro** [comment]
Please remain the json argument also.

**Comment by yuyu2172** [comment]
fixed

File: jsk_apc2016_common/node_scripts/visualize_pick_json.py [hunk 1]
```diff
@@ -21,14 +24,14 @@ def publish_cb(event):
 
     parser = argparse.ArgumentParser()
     parser.add_argument('json',
-                        help='JSON file with bin_contents and work_order')
+                        help='JSON file with bin_contents and work_order',
+                        required=False)
     args = parser.parse_args(rospy.myargv()[1:])
     json = args.json
```
Comment (wkentaro) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: if json is None:
  json = rospy.get_param(...)

is better?

**Comment by yuyu2172** [comment]
fixed

File: jsk_apc2016_common/node_scripts/visualize_pick_json.py [hunk 2]
```diff
@@ -11,6 +11,9 @@
 
 
 def publish_cb(event):
+    json = rospy.get_param('~json', None)
+    img = jsk_apc2016_common.visualize_pick_json(json)
+    imgmsg = br.cv2_to_imgmsg(img, encoding='bgr8')
```
Comment (wkentaro) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: Not necessary.

File: jsk_apc2016_common/node_scripts/visualize_pick_json.py [hunk 3]
```diff
@@ -21,14 +24,16 @@ def publish_cb(event):
...
+                        required=False)
     args = parser.parse_args(rospy.myargv()[1:])
     json = args.json
+    if json is None:
+        json = rospy.get_param('~json', None)
 
     img = jsk_apc2016_common.visualize_pick_json(json)
 
     br = cv_bridge.CvBridge()
-    imgmsg = br.cv2_to_imgmsg(img, encoding='bgr8')
```
Comment (wkentaro) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: Necessary.

File: jsk_apc2016_common/node_scripts/visualize_pick_json.py (hunk 2 above)
Comment (yuyu2172) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: They are necessary.

rosparam `json` gets dynamically changed. Output of this node has to reflect the change.

Comment (wkentaro) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: In that case, please use dynamic_reconfigure or use lower rate in line 38.

Comment (yuyu2172) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: ok

Comment (yuyu2172) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: fixed

File: jsk_apc2016_common/node_scripts/visualize_pick_json.py [hunk 4]
```diff
@@ -21,14 +24,16 @@ def publish_cb(event):
...
     parser.add_argument('json',
-                        help='JSON file with bin_contents and work_order')
+                        help='JSON file with bin_contents and work_order',
+                        required=False)
     args = parser.parse_args(rospy.myargv()[1:])
     json = args.json
+    if json is None:
+        json = rospy.get_param('~json', None)
 
     img = jsk_apc2016_common.visualize_pick_json(json)
```
Comment (wkentaro) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: not necessary

File: jsk_apc2016_common/node_scripts/visualize_pick_json.py [hunk 5]
```diff
@@ -21,14 +24,16 @@ def publish_cb(event):
...
 
     br = cv_bridge.CvBridge()
```
Comment (wkentaro) [inline jsk_apc2016_common/node_scripts/visualize_pick_json.py]: not necessary

**Comment by yuyu2172** [comment]
It now accepts json both by arg and by param.
Sorry for taking so long

**Comment by wkentaro** [comment]
I talked about this with @yuyu2172 and concluded this PR is not necessary.
He was troublesome to re-launch the `main.launch` because of the euslisp, but I said he can use `roslaunch jsk_2016_01_baxter_apc main.launch launch_main:=false json:=...`.

**Pull request closed without merging**